templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,7 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       if (obj.find(":selected").val() == '7' || obj.find(":selected").val() == '3') {
```

Comment:
I'm willing to merge this as-is because it's clearly an improvement.
But the "full" fix would probably be to assign the name/value pairs variable to the tpl so that javascript can use them to lookup values by name.